lib/ruby-debug-ide/xml_printer.rb

```diff
@@ -380,15 +386,46 @@ def max_compact_name_size
       50
     end
 
+    def inspect_with_allocation_control(slice)
+      x = Thread.current
+      
+      start_alloc_size = ObjectSpace.memsize_of_all
+      
+      trace = TracePoint.new(:c_call, :call) do |tp|
+        curr_alloc_size = ObjectSpace.memsize_of_all
+        
+        if(curr_alloc_size - start_alloc_size > 1e7)
+          
+          trace.disable
+          x.raise StandardError, "Out of memory: evaluation took longer than 10mb." if x.alive?
+        end
+        p [tp.path, tp.lineno, tp.event]
+      end
+
+      trace.enable
+      slice.inspect
+      trace.disable 
+    end
+    
+
     def compact_array_str(value)
       slice   = value[0..10]
-      compact = slice.inspect
+      
+      if defined?(JRUBY_VERSION)
+        compact = slice.inspect
+      else  
+        compact = inspect_with_allocation_control(slice)
+      end 
+      
       if value.size != slice.size
         compact[0..compact.size-2] + ", ...]"
       end
       compact
+    rescue StandardError => e
```

Comment:
I believe this relates to `inspect_with_allocation_control(slice)` call; I think it makes sense to place `rescue` there because in the other case it will be not clear why the thing is located here.